tracer/src/Datadog.Trace/OpenTelemetry/Logs/OtlpLogsSerializer.cs

```diff
@@ -0,0 +1,278 @@
+                    {
+                        continue;
+                    }
+
+                    writePosition = WriteResourceAttribute(buffer, writePosition, tag.Key, tag.Value);
+                }
+            }
+
+            return writePosition;
+        }
+
+        private static int WriteScopeLogs(byte[] buffer, int writePosition, IReadOnlyList<LogPoint> logs)
+        {
+            var scopeName = logs.Count > 0 && !StringUtil.IsNullOrEmpty(logs[0].CategoryName)
+                ? logs[0].CategoryName
+                : string.Empty;
+
+            writePosition = ProtobufSerializer.WriteTag(buffer, writePosition, ScopeLogs_Scope, ProtobufWireType.LEN);
+            int scopeLengthPosition = writePosition;
+            writePosition += ReserveSizeForLength;
+
+            writePosition = ProtobufSerializer.WriteStringWithTag(buffer, writePosition, InstrumentationScope_Name, scopeName ?? string.Empty);
+            writePosition = ProtobufSerializer.WriteStringWithTag(buffer, writePosition, InstrumentationScope_Version, string.Empty);
+
+            ProtobufSerializer.WriteReservedLength(buffer, scopeLengthPosition, writePosition - (scopeLengthPosition + ReserveSizeForLength));
+
+            for (int i = 0; i < logs.Count; i++)
+            {
+                writePosition = WriteLogRecord(buffer, writePosition, logs[i]);
+            }
+
+            writePosition = ProtobufSerializer.WriteStringWithTag(buffer, writePosition, ScopeLogs_Schema_Url, string.Empty);
+
+            return writePosition;
+        }
+
+        private static int WriteLogRecord(byte[] buffer, int writePosition, LogPoint log)
+        {
+            writePosition = ProtobufSerializer.WriteTag(buffer, writePosition, ScopeLogs_Log_Records, ProtobufWireType.LEN);
+            int logRecordLengthPosition = writePosition;
+            writePosition += ReserveSizeForLength;
+
+            var timeUnixNano = ConvertToUnixNano(log.Timestamp);
+            writePosition = ProtobufSerializer.WriteFixed64WithTag(buffer, writePosition, LogRecord_Time_Unix_Nano, timeUnixNano);
+            writePosition = ProtobufSerializer.WriteFixed64WithTag(buffer, writePosition, LogRecord_Observed_Time_Unix_Nano, timeUnixNano);
+            writePosition = ProtobufSerializer.WriteEnumWithTag(buffer, writePosition, LogRecord_Severity_Number, log.GetSeverityNumber());
+            writePosition = ProtobufSerializer.WriteStringWithTag(buffer, writePosition, LogRecord_Severity_Text, log.GetSeverityText());
+            writePosition = WriteLogRecordBody(buffer, writePosition, log.Message);
+
+            foreach (var attr in log.Attributes)
+            {
+                writePosition = WriteKeyValueAttribute(buffer, writePosition, attr.Key, attr.Value?.ToString() ?? string.Empty);
+            }
+
+            if (log.TraceId.HasValue && !log.TraceId.Value.ToString().Equals("00000000000000000000000000000000", StringComparison.OrdinalIgnoreCase))
```

Comment:
This seems janky 😅

Reply:
I recommended changing to using our Datadog trace-ids/span-ids, but if we were to keep using the `ActivityTraceId` type, I think you'd want this comparison

```suggestion
            if (log.TraceId != default(ActivityTraceId))
```